calculator.js

```diff
@@ -0,0 +1,32 @@
+var value = 0;
+
+var calc = {
+   add: function(add1, add2) {
+     value += add1 + add2;
+     return add1 + add2;
+   },
+   subtract: function(sub1, sub2) {
+     value +=  sub1 - sub2;
+     return sub1 - sub2;
+   },
+   multiply: function(mul1, mul2) {
+     value += mul1 * mul2;
+     return mul1 * mul2;
+   },
+   divide: function(div1, div2) {
+     value += div1 / div2;
+     return div1 / div2;
+   },
+   exponent: function(base, exponent) {
```

Comment:
Dig the hand-rolled approach. Check out Javascript's Math object. It has a method that takes care of exponents in one line. 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/pow